codegen/module.go

```diff
@@ -1362,49 +1367,35 @@ type yamlClassConfig struct {
 	IsExportGenerated *bool `yaml:"IsExportGenerated" json:"IsExportGenerated"`
 }
 
-// Read will read a class configuration yaml file into a yamlClassConfig struct
+// ConstructClassConfig unmarshals raw bytes into a ClassConfig struct
 // or return an error if it cannot be unmarshaled into the struct
-func (yamlConfig *yamlClassConfig) Read(
-	classConfigPath string,
-) ([]byte, error) {
-	configFile, readErr := ioutil.ReadFile(classConfigPath)
-	if readErr != nil {
-		return nil, errors.Wrapf(
-			readErr,
-			"Error reading class config %q",
-			classConfigPath,
-		)
-	}
-
-	parseErr := yaml.Unmarshal(configFile, &yamlConfig)
+func ConstructClassConfig(raw []byte, config *ClassConfig) error {
```

Comment:
nit: this constructor is awkward, shouldn't it construct `*ClassConfig` for you?

```
func NewClassConfig(bytes []byte) (*ClassConfig, error)
```